test/lib/tasks/generateLibraryManifest.js

```diff
@@ -0,0 +1,484 @@
+const {test} = require("ava");
+
+const ui5Builder = require("../../../");
+const tasks = ui5Builder.builder.tasks;
+const path = require("path");
+const ui5Fs = require("@ui5/fs");
+const resourceFactory = ui5Fs.resourceFactory;
+
+function createWorkspace() {
+	return resourceFactory.createAdapter({
+		virBasePath: "/",
+		project: {
+			metadata: {
+				name: "test.lib"
+			},
+			version: "2.0.0",
+			dependencies: [
+				{
+					metadata: {
+						name: "sap.ui.core"
+					},
+					version: "1.0.0"
+				}
+			]
+		}
+	});
+}
+
+function createDependencies() {
+	return resourceFactory.createAdapter({
+		fsBasePath: path.join(__dirname, "..", "..", "fixtures", "sap.ui.core-evo", "main", "src"),
+		virBasePath: "/resources"
+	});
+}
+
+async function assertCreatedManifest(t, oExpectedManifest) {
+	const {workspace, dependencies, resources} = t.context;
+
+	await Promise.all(resources.map((resource) => workspace.write(resource)));
+
+	await tasks.generateLibraryManifest({
+		workspace: workspace,
+		dependencies: dependencies,
+		options: {
+			projectName: "Test Lib"
+		}
+	});
+
+	const resource = await workspace.byPath("/resources/test/lib/manifest.json");
+	if (!resource) {
+		t.fail("Could not find /resources/test/lib/manifest.json in target");
+		return;
+	}
+
+	const buffer = await resource.getBuffer();
+	t.deepEqual(JSON.parse(buffer), oExpectedManifest, "Correct content");
+}
+
+test("Library without i18n bundle file", async (t) => {
+	t.context.workspace = createWorkspace();
+	t.context.dependencies = createDependencies();
+
+	t.context.resources = [];
+	t.context.resources.push(resourceFactory.createResource({
+		path: "/resources/test/lib/.library",
+		string: `
+			<?xml version="1.0" encoding="UTF-8" ?>
+			<library xmlns="http://www.sap.com/sap.ui.library.xsd" >
+
+				<name>test.lib</name>
+				<vendor>SAP SE</vendor>
+				<copyright></copyright>
+				<version>2.0.0</version>
+
+				<documentation>Test Lib</documentation>
+
+			</library>
+		`,
+		project: t.context.workspace._project
+	}));
+
+	await assertCreatedManifest(t, {
```

Comment:
much nicer testing :clap:

Reply:
I only added the relevant tests for the changed code, but I want to add more cases to cover the complete task / processor in here.

The only thing I don't like is passing the `project`, as I'm not sure it will be the same when using e.g. the CLI.